src/main/java/ch/njol/skript/conditions/CondIsDivisibleBy.java

```diff
@@ -0,0 +1,61 @@
+package ch.njol.skript.conditions;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.lang.Condition;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.util.Kleenean;
+import org.bukkit.event.Event;
+import org.jetbrains.annotations.Nullable;
+
+@Name("Divisible By")
+@Description("Check if a number is divisible by another number.")
+@Examples("5 is divisible by 5")
+@Since("INSERT VERSION")
+public class CondIsDivisibleBy extends Condition {
+
+	static {
+		Skript.registerCondition(CondIsDivisibleBy.class,
+			"%number% (is|can be) (divisible|divided) by %number%");
+	}
+
+	@SuppressWarnings("null")
+	private Expression<Number> num1;
+	@SuppressWarnings("null")
+	private Expression<Number> num2;
+
+	@Override
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		num1 = (Expression<Number>) exprs[0];
+		num2 = (Expression<Number>) exprs[1];
+		return true;
+	}
+
+	@Override
+	public boolean check(Event event) {
+		Number number1 = num1.getSingle(event);
+		Number number2 = num2.getSingle(event);
+
+		if (number1 == null || number2 == null) {
+			return false; // Prevent NullPointerException
+		}
+
+		double divisor = number2.doubleValue();
+
+		if (divisor == 0) {
+			return false; // Avoid division by zero
```

Comment:
i'm not sure this should return false as this is a mathematical error and so perhaps should also throw or at least log something. but idk

Reply:
then use mod function at this point, this is just a "hey this is true so it's correct" or "hey this is false so it isn't correct"

Reply:
yeah ig

Reply:
I don't think an error should be thrown here, **for now**. Until we have a system for runtime errors, this should just return false. Similarly issue in #7038